src/components/OrderBookWidget.tsx

```diff
@@ -81,12 +83,30 @@ const processData = (
 ): ProcessedItem[] => {
   let totalVolume = 0
   const isBid = type == Offer.Bid
+  // get lowest & average price of token vs token
+  // allows to get an idea of how to format numeros
+  const priceList = list.map(listItem => listItem.price)
+  const lowestPrice = Math.min(...priceList)
+
   return (
     list
       // Account fo decimals
       .map(element => {
+        const decimals = quoteToken.decimals - baseToken.decimals
+        // get the full native string amount to pass into parsedPrice
+        // as price comes in from API as number type
+        const price = (element.price / 10 ** decimals).toLocaleString([], { maximumSignificantDigits: 18 })
+        // return it as a BN via our parseAmount
+        const parsedPrice = parseAmount(price, 18)
```

Comment:
the precision is handled in the line above, that should suffice, no? the `parsedPrice` is only to get it into `BN` form before passing it into `smartFormat` - the precision there needs to be the same. Probably best to just wrap `parseAmount` into `smartFormat` to avoid needing to place precision twice